tests/test_term_analysis.cpp

```diff
@@ -0,0 +1,73 @@
+#include "gtest/gtest.h"
+#include "smt/available_solvers.h"
+#include "utils/term_analysis.h"
+
+using namespace pono;
+using namespace smt;
+using namespace std;
+
+namespace pono_tests {
+
+class TermAnalysisUnitTests : public ::testing::Test,
+                              public ::testing::WithParamInterface<SolverEnum>
+{
+ protected:
+  void SetUp() override
+  {
+    s = create_solver(GetParam());
+    boolsort = s->make_sort(BOOL);
+    bvsort = s->make_sort(BV, 8);
+    funsort = s->make_sort(FUNCTION, { bvsort, bvsort, bvsort });
+    relsort = s->make_sort(FUNCTION, { bvsort, bvsort, boolsort });
+  }
+  SmtSolver s;
+  Sort boolsort, bvsort, funsort, relsort;
+};
+
+TEST_P(TermAnalysisUnitTests, GetPredicates)
```

Comment:
should have a case for ITE

Reply:
Good idea -- added with 2d2e8719a4b54581d7961f391dba6d9d92fd2201. One question about that. If I have something like `x' = ite(x < 5, x+1, 0)`, would you expect the predicates to be `{x < 5 }` or `{x < 5, x' = ite(x < 5, x+1, 0)}`? I.e. does the equality between the term and the ITE-term count as a predicate? Currently it's the latter.

Reply:
We discussed offline and decided that the semantics of including the equality is correct, but it might be more performant to take the different cases: `x' = x+1` and `x' = 0`. This shouldn't be hard to implement so I will add this to the PR soon.